### PR TITLE
MM-31914 , MM-31898 - hook up license renewal button; update banners

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
1) hook up license renewal button in banner
2) Update on-prem banners to match Cloud banner style and sizing

The implementing of the solution to MM-31914 implied to show a cloud like styled banner, so it made more sense to modify at the same time all banners as requested in ticket MM-31898.

#### Ticket Link
[MM-31914](https://mattermost.atlassian.net/browse/MM-31914)
[MM-31898](https://mattermost.atlassian.net/browse/MM-31898)

#### Related Pull Requests
* has a related PR with mattermost-redux [PR1338](https://github.com/mattermost/mattermost-redux/pull/1338)

#### Screenshots
Show the new renewal license banner
Captura de pantalla 2021-01-20 a las 13 17 02

& hover state
Captura de pantalla 2021-01-20 a las 20 30 11

Shown when offline or when a problem with the renewal link
Captura de pantalla 2021-01-20 a las 13 17 39

Other banners examples
Captura de pantalla 2021-01-20 a las 13 16 13

Captura de pantalla 2021-01-20 a las 13 30 12

Captura de pantalla 2021-01-20 a las 13 36 45

Captura de pantalla 2021-01-20 a las 14 15 41

Captura de pantalla 2021-01-20 a las 20 28 20

### CODE REVIEW

File: components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        >
+            <g clipPath='url(#clip0)'>
+                <path
+                    d='M61.9647 216C96.1868 216 123.929 213.897 123.929 211.302C123.929 208.707 96.1868 206.604 61.9647 206.604C27.7425 206.604 0 208.707 0 211.302C0 213.897 27.7425 216 61.9647 216Z'
+                    fill='#3D3C40'
+                    fillOpacity='0.08'
+                />
+                <path
+                    d='M47.3647 47.9266C58.1484 43.348 71.5797 46.2489 84.6968 48.2623C84.8189 45.8241 86.2984 42.8567 84.7486 41.1556C82.8656 39.089 83.039 36.9175 84.172 34.7336C87.0694 29.1489 82.9183 23.2065 78.7269 18.2655C77.7822 17.1543 76.5882 16.2813 75.242 15.7172C73.8958 15.1532 72.4353 14.9141 70.9792 15.0193L63.2149 15.5729C61.3256 15.7076 59.5225 16.4151 58.047 17.6008C56.5715 18.7864 55.4939 20.3936 54.9584 22.2072C53.1044 24.7242 52.1122 27.2343 52.7369 29.7312C49.96 31.6256 49.4934 33.9123 50.7413 36.4901C51.7994 38.1132 51.7881 39.6932 50.6931 41.2294C49.2551 43.188 48.1537 45.3718 47.4341 47.6915L47.3647 47.9266Z'
+                    fill='#152235'
```
[the remark this answers is not in the document]
Comment (pablovelezvidal) [reply components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx:22]: the decision to put the svg in memory with the component is because when the modal was being shown for the offline situation, the image was being shown as a broken-image. By putting it in memory we make sure that will be available in offline mode.

**Comment by marianunez** [comment]
Adding @devinbinnie for apps platform visibility on the styling change for all announcement bars https://mattermost.atlassian.net/browse/MM-31898

File: components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (marianunez) [reply components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx:22]: Take a look at https://community-daily.mattermost.com/core/pl/jtiwrwc8mfff5ybc5tu5upq53w that shows how to make an svg theme compatible

File: components/announcement_bar/no_internet_connection/no_internet_connection.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    text-align: center;
+
+    .noInternetConnection__noAccessToInternet {
+        font-family: Open Sans;
+        font-style: normal;
+        font-weight: 600;
+        font-size: 20px;
+        line-height: 28px;
+        text-align: center;
+        color: #152235;
```
Comment (marianunez) [inline components/announcement_bar/no_internet_connection/no_internet_connection.scss:13]: Could we use any of the color vars for theme compatability here?

File: components/announcement_bar/no_internet_connection/no_internet_connection.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                <span className='noInternetConnection__contactSupport'>
+                    <FormattedMessage
+                        id='announcement_bar.warn.no_internet_connection'
+                        defaultMessage='In order to renew, you need to contact support.'
+                    />
+                </span>
+                <span className='noInternetConnection__emailUs'>
+                    <FormattedMarkdownMessage
+                        id='announcement_bar.warn.email_us'
+                        defaultMessage='Please [email Us](!{email}).'
```
Comment (marianunez) [inline components/announcement_bar/no_internet_connection/no_internet_connection.tsx:41]: Typo: 

```suggestion
                        defaultMessage='Please [email us](!{email}).'
```

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import NoInternetConnection from '../no_internet_connection/no_internet_connection';
+
+export interface RenewalLinkProps {
+    actions: {
+        openModal: (modalData: { modalId: string; dialogType: any; dialogProps?: any }) => void;
+    };
+}
+
+const RenewalLink: React.FC<RenewalLinkProps> = (props: RenewalLinkProps) => {
+    const [renewalLink, setRenewalLink] = useState('https://mattermost.com/renew/');
```
Comment (marianunez) [inline components/announcement_bar/renewal_link/renewal_link.tsx:23]: Not sure if we still need to default to this url, this is now being replaced with the new renewal link returned by the server.

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    useEffect(() => {
+        Client4.getRenewalLink().then(({renewal_link: renewalLinkParam}) => {
+            if (renewalLinkParam && (/^http[s]?:\/\/customers\.mattermost\.com\/subscribe\/renew/).test(renewalLinkParam)) {
```
Comment (marianunez) [inline components/announcement_bar/renewal_link/renewal_link.tsx:27]: Also, here we should not test against `https://customer.mattermost.com` given that this url is configurable in the server and this would break if updated in the future.

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                setRenewalLink(renewalLinkParam);
+            }
+        });
+    }, []);
+
+    const handleLinkClick = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.preventDefault();
+        trackEvent('announcement_bar', 'renew_license');
```
Comment (marianunez) [inline components/announcement_bar/renewal_link/renewal_link.tsx:35]: For the telemetry we usually group all related events in the same category, first param, and then the event name would be the action that was performed at the time, so based on https://docs.google.com/document/d/1HgZpqadMKvvSZdWoD_blysRwjdgqFbqn2o4zJNZGhYc/edit# I would suggest: 
```suggestion
        trackEvent('renew_license', 'renew_license_banner_success');
```

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        try {
+            const {status} = await Client4.ping();
+            if (status === 'OK') {
+                window.open(renewalLink, '_blank');
+            } else {
+                showModal();
```
Comment (marianunez) [inline components/announcement_bar/renewal_link/renewal_link.tsx:41]: nit: naming suggestion to provide more readability here

```suggestion
                showConnetionErrorModal();
```

File: i18n/en.json [hunk 8]
```diff
@@ -2207,6 +2207,10 @@
   "announcement_bar.warn_metric_status.number_of_posts.text": "You now have over 2,000,000 posts. We strongly advise using advanced features to avoid degraded performance.",
   "announcement_bar.warn_metric_status.number_of_users_ack.text": "Thank you for contacting Mattermost. We will follow up with you soon.",
   "announcement_bar.warn_metric_status.number_of_users.text": "You now have over 500 users. We strongly recommend using advanced features for large-scale servers.",
+  "announcement_bar.warn.contact_support": "In order to renew, you need to contact support.",
+  "announcement_bar.warn.email_us": "Please [email Us](!{email}).",
```
Comment (marianunez) [inline i18n/en.json:2211]:
```suggestion
  "announcement_bar.warn.email_us": "Please [email us](!{email}).",
```

**Review by marianunez — CHANGES_REQUESTED** [review]
Great stuff @pablovelezvidal! Some comments below

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+            }
+        } catch (error) {
+            showModal();
+        }
+    };
+
+    const showModal = () => {
```
Comment (marianunez) [inline components/announcement_bar/renewal_link/renewal_link.tsx:48]: We also need to add telemetry here for the failed link: 
```suggestion
    const showModal = () => {
            trackEvent('renew_license', 'renew_license_banner_fail');

```

**Comment by marianunez** [comment]
@pablovelezvidal we also need to update the redux hash [here](https://github.com/mattermost/mattermost-webapp/blob/dfb8860bd6a9005f4bed99cd7746d22f13abbb7b/package.json#L41) and in `package-lock.json` to point to the commit hash of your redux changes given that it's erroring out trying to resolve `Client4.getRenewalLink`

File: components/announcement_bar/renewal_link/renewal_link.tsx (hunk 7 above)
Comment (nickmisasi) [reply components/announcement_bar/renewal_link/renewal_link.tsx]: "showConnetionErrorModal" should be `showConnectionErrorModal` (missed the second c in "connection"

File: components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickmisasi) [reply components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx:22]: Spoke with Pablo today about this - putting the SVG into a component is fine, just need to fix up the theme styling so that it isn't jarring for dark themes

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        trackEvent('renew_license', 'renew_license_banner_success');
+        try {
+            const {status} = await Client4.ping();
+            if (status === 'OK' && renewalLink !== '') {
+                window.open(renewalLink, '_blank');
+            } else {
+                showConnetionErrorModal();
+            }
+        } catch (error) {
+            showConnetionErrorModal();
```
Comment (nickmisasi) [inline components/announcement_bar/renewal_link/renewal_link.tsx:44]:
```suggestion
            showConnectionErrorModal();
```

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+    const handleLinkClick = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        e.preventDefault();
+        trackEvent('renew_license', 'renew_license_banner_success');
+        try {
+            const {status} = await Client4.ping();
+            if (status === 'OK' && renewalLink !== '') {
+                window.open(renewalLink, '_blank');
+            } else {
+                showConnetionErrorModal();
```
Comment (nickmisasi) [inline components/announcement_bar/renewal_link/renewal_link.tsx:41]:
```suggestion
                showConnectionErrorModal();
```

File: components/announcement_bar/renewal_link/renewal_link.tsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        } catch (error) {
+            showConnetionErrorModal();
+        }
+    };
+
+    const showConnetionErrorModal = () => {
```
Comment (nickmisasi) [inline components/announcement_bar/renewal_link/renewal_link.tsx:48]:
```suggestion
    const showConnectionErrorModal = () => {
```

Comment (nickmisasi) [reply components/announcement_bar/renewal_link/renewal_link.tsx:48]: And any other occurrences :)

**Review by nickmisasi — CHANGES_REQUESTED** [review]

**Comment by nickmisasi** [comment]
Looks like some conflicts/type issues remain, I'd bet a branch update from master would fix majority of them

File: components/announcement_bar/no_internet_connection/no_internet_connection.tsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+                <span className='noInternetConnection__noAccessToInternet'>
+                    <FormattedMessage
+                        id='announcement_bar.warn.no_internet_connection'
+                        defaultMessage='Looks like you do not have access to the internet.'
+                    />
+                </span>
+                <span className='noInternetConnection__contactSupport'>
+                    <FormattedMessage
+                        id='announcement_bar.warn.contact_support'
+                        defaultMessage='In order to renew, you need to contact support.'
```
Comment (devinbinnie) [inline components/announcement_bar/no_internet_connection/no_internet_connection.tsx:35]: Just confirming, this isn't the bar that would pop on an on-prem installation correct? If so, this message might be confusing to them if their connection isn't working.

**Review by devinbinnie — APPROVED** [review]
LGTM other than Maria and Nick's comments. One non-blocking question of my own as well.

File: components/announcement_bar/no_internet_connection/no_internet_connection.tsx (hunk 13 above)
Comment (marianunez) [reply components/announcement_bar/no_internet_connection/no_internet_connection.tsx:35]: Agreed, brought this up in the channel [here](https://community-daily.mattermost.com/private-core/pl/sujnod5c7fyg7fu9tpwsaxd75o) and a new copy was suggested: 

```
It looks like you do not have access to the internet.

To renew your license, contact support at support@mattermost.com.

<Contact support>
```

File: components/announcement_bar/configuration_bar/configuration_bar.tsx [hunk 14]
```diff
@@ -157,148 +157,165 @@ class ConfigurationAnnouncementBar extends React.PureComponent<Props> {
...
-                return (
-                    <AnnouncementBar
-                        showCloseButton={true}
-                        handleClose={this.dismissExpiringLicense}
-                        type={AnnouncementBarTypes.ANNOUNCEMENT}
-                        message={
-                            <FormattedMarkdownMessage
-                                id={AnnouncementBarMessages.LICENSE_EXPIRING}
-                                defaultMessage='Enterprise license expires on {date, date, long}. [Please renew](!{link}).'
+        if (true) {
```
Comment (marianunez) [inline components/announcement_bar/configuration_bar/configuration_bar.tsx:206]: I believe this was an unintended change?

Comment (pablovelezvidal) [reply components/announcement_bar/configuration_bar/configuration_bar.tsx:206]: yes! we found it with @nickmisasi couple of minutes ago too!

File: components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickmisasi) [reply components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx:22]: @pablovelezvidal still need to handle the SVG themeing. Will approve once that's done

[the remark this answers is not in the document]
Comment (pablovelezvidal) [reply components/announcement_bar/no_internet_connection/no-internet-connection-svg.tsx:22]: If I understood correctly @alexsiclari the current design fits for light and dark themes

Captura de pantalla 2021-01-21 a las 19 32 23
Captura de pantalla 2021-01-21 a las 19 32 50

**Comment by marianunez** [comment]
@alexsiclari @stevemudie friendly reminder, this are ready for review

File: sass/layout/_sidebar-left.scss [hunk 15]
```diff
@@ -68,8 +68,24 @@
...
+        font-size: 12px;
+        font-weight: 600;
+        line-height: 18px;
+        letter-spacing: 0px;
+        text-align: center;
+        color: var(--mention-color);
+        border-radius: 9px;
+        background: var(--mention-bg);
+        -webkit-font-smoothing: subpixel-antialiased;
+        -moz-osx-font-smoothing: grayscale;
```
Comment (marianunez) [inline sass/layout/_sidebar-left.scss:88]: Is this change intended?

**Review by marianunez — APPROVED** [review]
LGTM, just a question below

**Comment by marianunez** [comment]
Could you provide the latest error dialog screenshot for UX review?

File: sass/layout/_sidebar-left.scss (hunk 15 above)
Comment (pablovelezvidal) [reply sass/layout/_sidebar-left.scss:88]: the original intention was just to remove the code related to the --tall-- banner. I am going to take a look!

**Review by stevemudie — APPROVED** [review]
LGTM

File: sass/layout/_sidebar-left.scss (hunk 15 above)
Comment (pablovelezvidal) [reply sass/layout/_sidebar-left.scss:88]: This commit was merged to master 4 days ago, https://github.com/mattermost/mattermost-webapp/pull/7357/files#diff-90fded696d50eef5af63163b66aaf08e3ee8190744429e674d8bad25d95a1e7fR74 it removed those lines from master. There should have been some merge conflict  that was not applied correctly and added these lines back as part of my fix. Great catch!

**Comment by pablovelezvidal** [comment]
> Could you provide the latest error dialog screenshot for UX review?
https://github.com/mattermost/mattermost-webapp/pull/7355#discussion_r562912732

**Comment by marianunez** [comment]
> > Could you provide the latest error dialog screenshot for UX review?
> > [#7355 (comment)](https://github.com/mattermost/mattermost-webapp/pull/7355#discussion_r562912732)

This one doesn't seem to include the updated texts.

File: sass/layout/_sidebar-left.scss (hunk 15 above)
Comment (marianunez) [reply sass/layout/_sidebar-left.scss]: Great, thanks!

**Comment by pablovelezvidal** [comment]
Captura de pantalla 2021-01-25 a las 22 53 14
Captura de pantalla 2021-01-25 a las 22 54 20

**Review by alexsiclari — APPROVED** [review]
@marianunez one small feedback change would be on the tooltip that is used on hovering of the banner renewal license. It is confusing that it says renew license now on the bottom, I would suggest removing the renew license text and just have inside the tooltip the license date information.

**Review by annelieseklein — APPROVED** [review]
Looks good! Agree with alex

**Comment by marianunez** [comment]
> @marianunez one small feedback change would be on the tooltip that is used on hovering of the banner renewal license. It is confusing that it says renew license now on the bottom, I would suggest removing the renew license text and just have inside the tooltip the license date information.

Agreed. Currently the general announcement bar sets the tooltip with the whole text of the bar. I've created a ticket to work on a fix for this outside of this PR. See https://mattermost.atlassian.net/browse/MM-32354
